crates/apub/src/objects/mod.rs

```diff
@@ -43,9 +48,15 @@ pub(crate) fn read_from_string_or_source_opt(
 /// wrapped in Announce. If we simply receive this like any other federated object, overwrite the
 /// existing, local Post. In particular, it will set the field local = false, so that the object
 /// can't be fetched from the Activitypub HTTP endpoint anymore (which only serves local objects).
-pub(crate) fn verify_is_remote_object(id: &Url, settings: &Settings) -> Result<(), LemmyError> {
-  let local_domain = settings.get_hostname_without_port()?;
-  if id.domain() == Some(&local_domain) {
+pub(crate) fn verify_is_remote_object<T>(
+  id: &ObjectId<T>,
+  context: &Data<LemmyContext>,
+) -> Result<(), LemmyError>
+where
+  T: activitypub_federation::traits::Object<DataType = LemmyContext> + Debug + Send + 'static,
+  for<'de2> <T as activitypub_federation::traits::Object>::Kind: serde::Deserialize<'de2>,
+{
+  if !id.is_local(context) {
```

Comment:
This is pretty hard to read because the fully qualified names are very long. I think importing `Object` (and `Deserialize` if you feel like it) will make this easier to understand. 

```rust
where
  T: Object<DataType = LemmyContext> + Debug + Send + 'static,
  for<'de2> <T as Object>::Kind: Deserialize<'de2>,
```